ostd/src/trap/handler.rs

```diff
@@ -1,7 +1,23 @@
 // SPDX-License-Identifier: MPL-2.0
 
+use alloc::boxed::Box;
+
+use spin::Once;
+
 use crate::{arch::irq::IRQ_LIST, cpu_local_cell, trap::TrapFrame};
 
+static BOTTOM_HALF_HANDLER: Once<Box<dyn Fn() + Sync + Send>> = Once::new();
+
+/// Register a function to the interrupt bottom half execution.
+///
+/// The function can only be registered once; subsequent calls will do nothing.
```

Comment:
Explain more about it. What is so-called "bottom half"? When will the registered handler be called?